### PR TITLE
Make unacceptable accuracy threshold configurable

### LINKED ISSUE #4961: Make unacceptable geopoint accuracy cutoff configurable
```
 category = if accuracy > unnacceptableAccuracyThreshold
    unacceptable
else if accuracy > accuracyThreshold + 5
    poor
else
    improving
```
<details>
  <summary>Brainstorming</summary>
Currently there is a accuracyThreshold body attribute: “For <input> with type geopoint, geotrace, or geoshape this sets the auto-accept threshold in meters for geopoint captures”. I’m not entirely sure how/if this is actually supported for traces and shapes. 

We want to also make the unacceptable accuracy cutoff configurable. I think by analogy to `accuracyThreshold` it should be a body attribute in camel case. Additionally, I think we should put it in the http://www.opendatakit.org/xforms namespace. For now it should only apply to manual geopoint.  

I propose `odk:worstAcceptableAccuracy`

Note: I think these are appropriate as body attributes rather than bind attributes because they're more like user experience concerns rather than data concerns. In the case of this new attribute, it can be entirely ignored by a client that doesn't have the new dialog with no data difference.

I think implementation can happen even if we’re not totally sure about the name since it would be easy enough to change. What users will really interact with will be XLSForm which we can use a different name for. 

If the attribute is set, its value is used to determine when to go past the unacceptable accuracy step in the geopoint dialog. If it’s not set, 100m is used. If it is set but is less than accuracyThreshold or accuracyThreshold + 5, then the unacceptable accuracy step on the progress bar is always filled in and the text never is displayed. 
<details>

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4961

#### What has been done to verify that this works as intended?

New tests.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

It'd be good to test the geo widgets but nothing else has really been touched.

#### Do we need any specific form for testing your changes? If so, please attach one.

Any form with geo widgets involved.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.

https://github.com/getodk/docs/issues/1422

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/FormEntryPromptUtils.java
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/FormEntryPromptUtils.java]: Maybe getBodyAttribute to be even more specific

File: collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ActivityGeoDataRequester.kt
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ActivityGeoDataRequester.kt:21]: I looked at converting the test as well but backed out when I didn't get a clean auto convert - don't want to hold this up or introduce more risk than we need to.

File: collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.java
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.java]: Remove comment

File: collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.java
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.java:73]: `acc` doesn't make sense as var name anymore

File: geo/src/main/java/org/odk/collect/geo/GeoPointViewModel.kt
Comment (lognaturel) [inline geo/src/main/java/org/odk/collect/geo/GeoPointViewModel.kt]: Artifact of another direction you considered?

File: geo/src/main/java/org/odk/collect/geo/GeoPointViewModel.kt
Comment (lognaturel) [inline geo/src/main/java/org/odk/collect/geo/GeoPointViewModel.kt]: This is a very interesting way to represent the way the accuracy is classified! The enum it looks like you considered would have been more like what I would have reached for but I see the advantages of this approach.

File: collect_app/src/main/java/org/odk/collect/android/utilities/FormEntryPromptUtils.java
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/FormEntryPromptUtils.java]: World's tiniest nit: `acc` doesn't make any sense anymore (it used to mean accuracy).

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Will let you decide if you want to fix the var name before merge.

Feels very low risk. Should we merge and do another beta so @getodk/testers can verify as part of regression testing?

**Comment by seadowg** [comment]
> Should we merge and do another beta so @getodk/testers can verify as part of regression testing?

Sounds good. I haven't had a chance to validate this with an example form, yet so let's do that first if we're going to merge and release without QA. Let me know if you've already given it a try.

**Comment by seadowg** [comment]
OK I've given it a try so going to merge now!

**Pull request merged**
